service/src/main/java/tds/session/repositories/impl/SessionRepositoryImpl.java

```diff
@@ -0,0 +1,46 @@
+package tds.session.repositories.impl;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.IncorrectResultSizeDataAccessException;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.jdbc.core.namedparam.SqlParameterSource;
+import org.springframework.stereotype.Repository;
+
+import javax.sql.DataSource;
+import java.util.Optional;
+import java.util.UUID;
+
+import tds.common.data.mysql.UuidAdapter;
+import tds.common.data.mysql.spring.UuidBeanPropertyRowMapper;
+import tds.session.Session;
+import tds.session.repositories.SessionRepository;
+
+@Repository
+public class SessionRepositoryImpl implements SessionRepository {
+    private final NamedParameterJdbcTemplate jdbcTemplate;
+
+    @Autowired
+    public SessionRepositoryImpl(DataSource datasource) {
+        this.jdbcTemplate = new NamedParameterJdbcTemplate(datasource);
+    }
+
+    @Override
+    public Optional<Session> getSessionById(UUID id) {
+        final SqlParameterSource parameters = new MapSqlParameterSource("id", UuidAdapter.getBytesFromUUID(id));
+
+        String query = "SELECT _Key as id, sessionType as type, status \n" +
+                       "FROM session.session \n" +
+                       "WHERE _Key = :id";
+
+        Optional<Session> sessionOptional;
+        try {
+            sessionOptional = Optional.of(jdbcTemplate.queryForObject(query, parameters, new UuidBeanPropertyRowMapper<>(Session.class)));
+        } catch (IncorrectResultSizeDataAccessException e) {
```

Comment:
No, we do not need to add a catch for `EmptyResultDataAccessException` because it extends `IncorrectResultSizeDataAccessException` so the try/catch that exists will catch the `EmptyResultDataAccessException` if thrown.  

All that being said this does raise an interesting question.  We know due to the DB that we should never have more than one record for a given ID.  If for some reason we do get multiple records back for a given ID it would mean something is wrong with the DB configuration so maybe catching the narrower Exception is what we should do so when there are multiple records it'll bubble and log a 500.

**Spring's code below**
```java
public static <T> T requiredSingleResult(Collection<T> results) throws IncorrectResultSizeDataAccessException {
        int size = results != null?results.size():0;
        if(size == 0) {
            throw new EmptyResultDataAccessException(1);
        } else if(results.size() > 1) {
            throw new IncorrectResultSizeDataAccessException(1, size);
        } else {
            return results.iterator().next();
        }
    }